src/prefect/futures.py

```diff
@@ -292,6 +292,59 @@ def result(
             ) from exc
 
 
+def as_completed(futures: List[PrefectFuture], timeout=None) -> Iterator[PrefectFuture]:
```

Comment:
i'd double check my suggestion, but this looks like its a Generator function right?
```suggestion
def as_completed(futures: List[PrefectFuture], timeout: Optional[float] = None) -> Generator[PrefectFuture, None]:
```